### PR TITLE
Add ML perf files

### PR DESCRIPTION
**Pull request by benharsh**
I think the main difference from the old approach is that SSCA2 size22 is now under studies/ssca2/bharshbarg, as to not clutter the release directory  even more.

### CODE REVIEW

**Comment by benharsh** [comment]
@chapel-lang/perf-team : any takers?

File: test/release/examples/benchmarks/miniMD/miniMD.skipif [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env python
+
+# Skip this test for gasnet+mpi performance testing. As of 08/08/15 it takes
+# ~30 minutes to complete
+import os
+
+print(os.getenv('CHPL_TEST_PERF') == 'on' and
+      os.getenv('CHPL_COMM_SUBSTRATE') == 'mpi')
```
Comment (ronawho) [inline test/release/examples/benchmarks/miniMD/miniMD.skipif]:
```python
print(os.getenv('CHPL_TEST_PERF') == 'on' and
      os.getenv('CHPL_COMM_SUBSTRATE') == 'mpi')
```

=>

```python
print(os.getenv('CHPL_TEST_PERF_LABEL') == 'ml-' and
      os.getenv('CHPL_TEST_PERF') == 'on' and
      os.getenv('CHPL_COMM_SUBSTRATE') == 'mpi')
```

just to make it a little more clear that the skipif is really meant for multi-locale testing?

File: test/studies/ssca2/bharshbarg/SSCA2_main.size22.skipif [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env python
+
+# Skip size22 for gasnet+mpi and gasnet+aries performance testing. As of
+# 08/08/15 it takes over 30 minutes to complete. This can be re-evaluated as
+# size 16 gets faster.
+import os
+
+print(os.getenv('CHPL_TEST_PERF') != 'on' or (os.getenv('CHPL_TEST_PERF') == 'on' and
+      os.getenv('CHPL_COMM_SUBSTRATE') in ['mpi', 'aries']))
```
Comment (ronawho) [inline test/studies/ssca2/bharshbarg/SSCA2_main.size22.skipif]:
```python
print(os.getenv('CHPL_TEST_PERF') != 'on' or (os.getenv('CHPL_TEST_PERF') == 'on' and
      os.getenv('CHPL_COMM_SUBSTRATE') in ['mpi', 'aries']))
```
=>

```python


print(os.getenv('CHPL_TEST_PERF_LABEL') != 'ml-' or
      os.getenv('CHPL_COMM') == 'gasnet')
```

?

**Comment by ronawho** [comment]
I think we can remove all the SSCA2_main.size22 crap. 

Think you just have to do

```bash
git mv $CHPL_HOME/test/studies/ssca2/bharshbarg/SSCA2_main.size22.ml-timeout $CHPL_HOME/test/release/examples/benchmarks/ssca2/SSCA2_main.ml-timeout 
```

and change SSCA2_main.ml-execopts to be:

```python
#!/usr/bin/env python

import os

# Only run size22 under ugni. As of 08/08/15 gasnet+{mpi, aries} performance
# testing takes over 30 minutes to complete
size16 = '--SCALE=16 --LOW_APPROX_SCALE=4 --TOP_APPROX_SCALE=4  # SSCA2-16.4'
size22 = '--SCALE=22 --LOW_APPROX_SCALE=4 --TOP_APPROX_SCALE=4  # SSCA2-22.4'

print(size16)

if os.getenv('CHPL_COMM') == 'ugni':
    print(size22)
```

and make sure to make the execopts executable.

Then I think all the size22 stuff can be removed. Should be nice for minimizing files, and has the added bonus of not having to recompile SSCA2 for ugni

**Comment by ronawho** [comment]
And I'm like 90% sure that executable execopts thing will work. I'm heading out for the day, but will try to remember to test it tomorrow

**Comment by ronawho** [comment]
Overall, looks good to me 👍 


After this has been in for a few days I think we can remove the .ml-numlocales files and just throw `--numlocales 16` to start_test in the perf test script.

I think it might also be worth reviewing the execenv files to see if we really need to be setting the things we are. In particular I'm pretty sure we don't need to be setting the call stack to 1M for stream global/promoted anymore. This isn't super important, but should help minimize the number of files

**Comment by benharsh** [comment]
Testing looks good for my recent commit. Good call on the execopts!

**Comment by ronawho** [comment]
👍

**Pull request merged**